tools/mason/MasonExternal.chpl

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright 2004-2018 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+use MasonUtils;
+use FileSystem;
+use MasonHelp;
+use MasonEnv;
+use Path;
+use TOML;
+
+proc masonExternal(args: [] string) {
+  try! {
+    if args.size < 3 {
+      masonExternalHelp();
+      exit(0);
+    }
+    else if args[2] == "--setup" {
+      setupSpack();
+      exit(0);
+    }
+    if spackInstalled() {
+      select (args[2]) {
+        when 'search' do searchSpkgs(args);
+        when 'compiler' do compiler(args);
+        when 'install' do installSpkg(args);
+        when 'uninstall' do uninstallSpkg(args);
+        when 'info' do spkgInfo(args);
+        when 'find' do findSpkg(args);
+        when '--spec' do specHelp();
+        when '--help' do masonExternalHelp();
+        when '-h' do masonExternalHelp();
+        otherwise {
+          writeln('error: no such subcommand');
+          writeln('try mason external --help');
+          exit(1);
+        }
+      }
+    }
+  }
+  catch e: MasonError {
+    writeln(e.message());
+    exit(1);
+  }
+}
+
+private proc specHelp() {
+  const command = "spack help --spec";
+  const status = runSpackCommand(command);
+}
+
+private proc spackInstalled() throws {
+  if !isDir(MASON_HOME + "/spack") {
+    throw new MasonError("To use `mason external` call mason external --setup");
+  }
+  return true;
+}
+
+private proc setupSpack() throws {
+  writeln("Installing Spack backend ...");
+  const spackVers = "releases/v0.11.2";
```

Comment:
seems like something that should be in a const/config const at the top of the file

Reply:
You can parse args manually and support config consts (there is a test somewhere showing how, ask me if you can't find it)

Reply:
But I agree - it's more important that it's called out at the top of the file than that it's settable with a command line argument.